src/SFML/Audio/SoundEffect.cpp

```diff
@@ -0,0 +1,310 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2020 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_SYSTEM_IOS
+
+#include <SFML/Audio/AudioDevice.hpp>
+#include <SFML/Audio/SoundEffect.hpp>
```

Comment:
> but the CI doesn't seem to recognise the SFML_SYSTEM_IOS definition from Config.hpp?

Took me a while to figure this out, but since you're including the SoundEffect header only inside the `#ifndef` block, `SFML_SYSTEM_IOS` won't be defined at the time you check it, as the C++ file is empty from the perspective of the pre-processor when reaching that line.
Meaning, you should move this line outside the `#ifndef` blocks.

Reply:
🤦‍♀️ Next question is: as this breaks the style guide for alphabetical includes (trying to follow it is how I managed to introduce this problem in the first place) can an exception be made in this specific case?

Reply:
Well yeah, if there's a technical reason why stuff has to be different, then it's okay to deviate from the guidelines.

Reply:
No problem, just didn't want to make assumptions.